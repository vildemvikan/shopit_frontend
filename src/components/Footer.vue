<script setup lang="ts">
import { ref, watch } from 'vue'
import { useI18n } from 'vue-i18n'
import { displayStore } from '@/stores/displayStore.ts'
const { t, locale } = useI18n()

const display = displayStore()

const theme = ref<'manual-light' | 'manual-dark'>('manual-light')

function toggleTheme(newTheme: 'manual-light' | 'manual-dark') {
  display.updateMode(newTheme)
  theme.value = newTheme
}

function changeLanguage(lang: string) {
  display.updateLanguage(lang)
  locale.value = lang
}

watch(theme, (newTheme) => {
  const root = document.documentElement
  root.classList.remove('manual-light', 'manual-dark')
  root.classList.add(newTheme)
})
</script>

<template>
  <div class="content">

    <div class="modes">
      <div class="mode" id="dark-mode">
        <img src="@/assets/icons/dark.svg"
             alt="dark mode"
             class="icon"
             @click="toggleTheme('manual-dark')"
        >
      </div>
      <div class="mode" id="light">
        <img src="@/assets/icons/light.svg"
             alt="light mode"
             class="icon"
             @click="toggleTheme('manual-light')"
        >
      </div>
    </div>
    <div class="languages">
      <img src="@/assets/icons/norwegian.svg.webp"
           v-if="locale == 'no'"
           alt="norwegian"
           class="icon"
           @click="changeLanguage('en')"
      >
      <img src="@/assets/icons/english.svg.webp"
           v-if="locale == 'en'"
           alt="english"
           class="icon"
           @click="changeLanguage('no')"
      >
    </div>

  </div>

</template>

<style scoped>
.content{
  display: flex;
  flex-direction: row;
  height: 100%;
  width: 100%;
  background-color: var(--color-lavendel-background);
  align-items: center;
  place-content: end;
  gap: 20px;
  padding: 0 20px;
}

.languages{
  display: flex;
  flex-direction: row;
  height: 50%;
}

.modes{
  display: flex;
  flex-direction: row;
  height: 80%;
}

.mode{
  height: 100%;
  aspect-ratio: 1/1;
  padding: 5px;
}

#dark-mode{
  border-right: 2px solid black;
}

.icon{
  height: 100%;
}

.icon:hover{
  transform: scale(1.05);
  cursor: pointer;
}

</style>
