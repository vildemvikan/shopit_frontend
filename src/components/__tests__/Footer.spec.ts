// Footer.spec.ts
import { describe, it, expect, vi, beforeEach } from 'vitest'
import { mount } from '@vue/test-utils'
import Footer from '@/components/Footer.vue'

// Mock displayStore
vi.mock('@/stores/displayStore.ts', () => {
  return {
    displayStore: () => ({
      updateMode: vi.fn(),
      updateLanguage: vi.fn()
    })
  }
})

// Mock i18n
const mockLocale = { value: 'en' }
vi.mock('vue-i18n', () => ({
  useI18n: () => ({
    t: (key: string) => key,
    locale: mockLocale
  })
}))

describe('Footer.vue', () => {
  let wrapper: ReturnType<typeof mount>

  beforeEach(() => {
    wrapper = mount(Footer)
  })

  it('should render and match snapshot', () => {
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('toggles to dark mode on dark icon click', async () => {
    const darkIcon = wrapper.find('#dark-mode .icon')
    await darkIcon.trigger('click')
    expect(document.documentElement.classList.contains('manual-dark')).toBe(true)
  })

  it('toggles to light mode on light icon click', async () => {
    const lightIcon = wrapper.find('#light .icon')
    await lightIcon.trigger('click')
    expect(document.documentElement.classList.contains('manual-light')).toBe(false)
  })
})
