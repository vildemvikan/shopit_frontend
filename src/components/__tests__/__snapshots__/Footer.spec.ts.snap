// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Footer.vue > should render and match snapshot 1`] = `
"<div data-v-4896eafd="" class="content">
  <div data-v-4896eafd="" class="modes">
    <div data-v-4896eafd="" class="mode" id="dark-mode"><img data-v-4896eafd="" src="data:image/svg+xml,%3c?xml%20version='1.0'%20?%3e%3c!--%20Uploaded%20to:%20SVG%20Repo,%20www.svgrepo.com,%20Generator:%20SVG%20Repo%20Mixer%20Tools%20--%3e%3csvg%20fill='%23000000'%20width='800px'%20height='800px'%20viewBox='0%200%2035%2035'%20data-name='Layer%202'%20id='Layer_2'%20xmlns='http://www.w3.org/2000/svg'%3e%3cpath%20d='M18.44,34.68a18.22,18.22,0,0,1-2.94-.24,18.18,18.18,0,0,1-15-20.86A18.06,18.06,0,0,1,9.59.63,2.42,2.42,0,0,1,12.2.79a2.39,2.39,0,0,1,1,2.41L11.9,3.1l1.23.22A15.66,15.66,0,0,0,23.34,21h0a15.82,15.82,0,0,0,8.47.53A2.44,2.44,0,0,1,34.47,25,18.18,18.18,0,0,1,18.44,34.68ZM10.67,2.89a15.67,15.67,0,0,0-5,22.77A15.66,15.66,0,0,0,32.18,24a18.49,18.49,0,0,1-9.65-.64A18.18,18.18,0,0,1,10.67,2.89Z'/%3e%3c/svg%3e" alt="dark mode" class="icon"></div>
    <div data-v-4896eafd="" class="mode" id="light"><img data-v-4896eafd="" src="data:image/svg+xml,%3c?xml%20version='1.0'%20encoding='utf-8'?%3e%3c!--%20Uploaded%20to:%20SVG%20Repo,%20www.svgrepo.com,%20Generator:%20SVG%20Repo%20Mixer%20Tools%20--%3e%3csvg%20width='800px'%20height='800px'%20viewBox='0%200%2024%2024'%20fill='none'%20xmlns='http://www.w3.org/2000/svg'%3e%3cg%20clip-path='url(%23a)'%20fill='%23000000'%3e%3cpath%20d='M12%200a1%201%200%200%201%201%201v4a1%201%200%201%201-2%200V1a1%201%200%200%201%201-1ZM4.929%203.515a1%201%200%200%200-1.414%201.414l2.828%202.828a1%201%200%200%200%201.414-1.414L4.93%203.515ZM1%2011a1%201%200%201%200%200%202h4a1%201%200%201%200%200-2H1ZM18%2012a1%201%200%200%201%201-1h4a1%201%200%201%201%200%202h-4a1%201%200%200%201-1-1ZM17.657%2016.243a1%201%200%200%200-1.414%201.414l2.828%202.828a1%201%200%201%200%201.414-1.414l-2.828-2.828ZM7.757%2017.657a1%201%200%201%200-1.414-1.414L3.515%2019.07a1%201%200%201%200%201.414%201.414l2.828-2.828ZM20.485%204.929a1%201%200%200%200-1.414-1.414l-2.828%202.828a1%201%200%201%200%201.414%201.414l2.828-2.828ZM13%2019a1%201%200%201%200-2%200v4a1%201%200%201%200%202%200v-4ZM12%207a5%205%200%201%200%200%2010%205%205%200%200%200%200-10Z'/%3e%3c/g%3e%3cdefs%3e%3cclipPath%20id='a'%3e%3cpath%20fill='%23ffffff'%20d='M0%200h24v24H0z'/%3e%3c/clipPath%3e%3c/defs%3e%3c/svg%3e" alt="light mode" class="icon"></div>
  </div>
  <div data-v-4896eafd="" class="languages">
    <!--v-if-->
    <!--v-if-->
  </div>
</div>"
`;
